.github/workflows/main.yml

```diff
@@ -22,7 +22,8 @@ jobs:
       - name: Lint
         run: |
           yarn run lint --no-fix
-          DEBUG="lockfile-lint,validate-host-manager" npx lockfile-lint --path yarn.lock --validate-https --allowed-hosts npm --allowed-hosts registry.yarnpkg.com
+          # blocked by https://github.com/lirantal/lockfile-lint/issues/101
+          # DEBUG="lockfile-lint,validate-host-manager" npx lockfile-lint --path yarn.lock --validate-https --allowed-hosts npm --allowed-hosts registry.yarnpkg.com
```

Comment:
You can use `networkSettings` to disallow everything except the registry, http is blocked by default
Example here: https://github.com/yarnpkg/berry/pull/2030#issuecomment-716854341

Reply:
Thanks @merceyz! That seems to cover the `--allowed-hosts` feature of `lockfile-lint`.

I had a look at the Yarn docs, and found `enableStrictSsl`. I think that's equivalent to `lockfile-lint`'s `--validate-https`. 

```diff
diff --git a/.github/workflows/main.yml b/.github/workflows/main.yml
index bc356864..793214c1 100644
--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -22,8 +22,6 @@ jobs:
       - name: Lint
         run: |
           yarn run lint --no-fix
-          # blocked by https://github.com/lirantal/lockfile-lint/issues/101
-          # DEBUG="lockfile-lint,validate-host-manager" npx lockfile-lint --path yarn.lock --validate-https --allowed-hosts npm --allowed-hosts registry.yarnpkg.com
       - name: Test
         run: |
           yarn run coverage:unit
diff --git a/.yarnrc.yml b/.yarnrc.yml
index 90e282d0..2b422cd3 100644
--- a/.yarnrc.yml
+++ b/.yarnrc.yml
@@ -1,3 +1,12 @@
 nodeLinker: node-modules
 
 yarnPath: .yarn/releases/yarn-3.0.0-rc.2.cjs
+
+enableNetwork: false
+enableStrictSsl: true
+
+networkSettings:
+  "registry.yarnpkg.com":
+    enableNetwork: true
+  "registry.npmjs.org":
+    enableNetwork: true
```

Removed `node_modules`, then `yarn install`'ed, and it worked fine ! :tada: 

Then edited the `networkSettings` host names (adding extra letters to make the domain names invalid).

```diff
diff --git a/.yarnrc.yml b/.yarnrc.yml
index 2b422cd3..340a5206 100644
--- a/.yarnrc.yml
+++ b/.yarnrc.yml
@@ -6,7 +6,7 @@ enableNetwork: false
 enableStrictSsl: true
 
 networkSettings:
-  "registry.yarnpkg.com":
+  "registry.ayarnpkg.com":
     enableNetwork: true
-  "registry.npmjs.org":
+  "registry.anpmjs.org":
     enableNetwork: true
```

But `rm -rf node_modules && yarn install` worked fine again? @merceyz, shouldn't that have failed?

Reply:
Pushed the latest code, minus the last diff with the typo in the URL.

```bash
kinow@ranma:~/Development/python/workspace/cylc-ui$ yarn -v
3.0.0-rc.2

```

Reply:
> I had a look at the Yarn docs, and found enableStrictSsl. I think that's equivalent to lockfile-lint's --validate-https.

That is enabled by default and causes Yarn to throw if there is an https errors

> But rm -rf node_modules && yarn install worked fine again? @merceyz, shouldn't that have failed?

No, it doesn't make any network requests since you had your lockfile and a fully populated cache. You can run `yarn add lodash` to see it throw.

Reply:
Ah, you are correct. I deleted the cache directories (in `~` and in `.yarn/cache`) and got

```
➤ YN0001: │ Error: zen-observable@npm:0.8.15: Request to 'https://registry.yarnpkg.com/zen-observable/-/zen-observable-0.8.15.tgz' has been blocked because of your configuration settings
    at Tu (/home/kinow/Development/python/workspace/cylc-ui/.yarn/releases/yarn-3.0.0-rc.2.cjs:237:2702)
    at /home/kinow/Development/python/workspace/cylc-ui/.yarn/releases/yarn-3.0.0-rc.2.cjs:237:3692
    at Object.getFactoryWithDefault (/home/kinow/Development/python/workspace/cylc-ui/.yarn/releases/yarn-3.0.0-rc.2.cjs:10:29890)
    at Object.Iue (/home/kinow/Development/python/workspace/cylc-ui/.yarn/releases/yarn-3.0.0-rc.2.cjs:237:3658)
    at Ln (/home/kinow/Development/python/workspace/cylc-ui/.yarn/releases/yarn-3.0.0-rc.2.cjs:527:13112)
    at Mn.fetchFromNetwork (/home/kinow/Development/python/workspace/cylc-ui/.yarn/releases/yarn-3.0.0-rc.2.cjs:527:19123)
    at loader (/home/kinow/Development/python/workspace/cylc-ui/.yarn/releases/yarn-3.0.0-rc.2.cjs:527:18935)
    at u (/home/kinow/Development/python/workspace/cylc-ui/.yarn/releases/yarn-3.0.0-rc.2.cjs:281:1042)
    at async h (/home/kinow/Development/python/workspace/cylc-ui/.yarn/releases/yarn-3.0.0-rc.2.cjs:281:1522)
    at async f (/home/kinow/Development/python/workspace/cylc-ui/.yarn/releases/yarn-3.0.0-rc.2.cjs:281:2283)
➤ YN0000: └ Completed in 4s 79ms
➤ YN0000: Failed with errors in 4s 504ms
```

Reply:
Removed the TODO, and the `networkSettings` is working now. Thanks @merceyz !